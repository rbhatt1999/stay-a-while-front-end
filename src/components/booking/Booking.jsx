import { useEffect, useState } from 'react';
import { FcCalendar } from 'react-icons/fc';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import BookingForm from './BookingForm';
import styles from './Booking.module.css';
import { fetchHomeStays } from '../home/HomeSlice';

export default function Booking() {
  const [booking, setBooking] = useState(false);
  const homeStayList = useSelector((state) => state.home.listings);
  const dispatch = useDispatch();
  const { id } = useParams();
  useEffect(() => {
    if (homeStayList.length === 0) {
      const token = localStorage.getItem('token');
      dispatch(fetchHomeStays(token));
    }
    if (id) {
      setBooking(true);
    }
  }, [dispatch, homeStayList, id]);
  return (
    <>
      <section className={styles.booking_section}>
        <div className="booking">
          <h1>Book A Home with Stay A While</h1>
          <hr />
        </div>
        <p className={styles.welcome_message}>
          Welcome to Stay A While, where we believe that every traveler
          deserves a home away from home.
          We pride ourselves on creating a homey feeling that sets us apart from
          other vacation rental platforms.
          Our hosts are dedicated to making your stay as comfortable and enjoyable as
          possible, and are always available to assist you with any questions or concerns.
          So why not book a stay with us and experience the comfort and warmth
          of a home away from home?
          We guarantee you won&apos;t be disappointed.
        </p>
        <span>
          <button
            type="submit"
            onClick={() => setBooking(true)}
            className={styles.booking_button}
          >
            <FcCalendar />
            {' '}
            <span>Book Now</span>
          </button>
        </span>
        {booking && (
        <div className="flex items-center justify-center w-full h-full absolute top-0 left-0 z-50">
          <div
            className={styles.booking_form}
          >
            <BookingForm onClose={() => setBooking(false)} />
          </div>
          <div className={styles.backdrop} />
        </div>
        )}
      </section>
    </>
  );
}
