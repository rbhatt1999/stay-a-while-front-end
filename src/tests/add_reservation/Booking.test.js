import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import {
  render, act, screen,
} from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import axios from 'axios';
import homestays from '../delete_homeStay/home_stays.json';
import homeSlice from '../../components/home/HomeSlice';
import Booking from '../../components/booking/Booking';

jest.mock('axios');

const store = configureStore({
  reducer: {
    home: homeSlice.reducer,
  },
});
describe('Booking', () => {
  beforeEach((async () => {
    await act(async () => {
      render(
        <Provider store={store}>
          <Booking />
        </Provider>,
      );
    });
    axios.get.mockResolvedValue({
      data: {
        homeStayList: homestays.data.home_stays,
      },
    });

    await act(async () => {
      render(
        <Provider store={store}>
          <Booking />
        </Provider>,
      );
    });
  }));

  test('should have a submit button', () => {
    expect(screen.getAllByRole('button', { name: /Book Now/i })).toHaveLength(2);
  });

  test('should have a h1 element with text Book A Home with Stay A While', () => {
    expect(screen.getAllByText(/Book A Home with Stay A While/i)).toHaveLength(2);
  });

  test('should match screenshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Booking />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
