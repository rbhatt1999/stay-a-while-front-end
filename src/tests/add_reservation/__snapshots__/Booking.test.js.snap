// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking should match screenshot 1`] = `
<section
  className="booking_section"
>
  <div
    className="booking"
  >
    <h1>
      Book A Home with Stay A While
    </h1>
    <hr />
  </div>
  <p
    className="welcome_message"
  >
    Welcome to Stay A While, where we believe that every traveler deserves a home away from home. We pride ourselves on creating a homey feeling that sets us apart from other vacation rental platforms. Our hosts are dedicated to making your stay as comfortable and enjoyable as possible, and are always available to assist you with any questions or concerns. So why not book a stay with us and experience the comfort and warmth of a home away from home? We guarantee you won't be disappointed.
  </p>
  <span>
    <button
      className="booking_button"
      onClick={[Function]}
      type="submit"
    >
      <svg
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5,38V14h38v24c0,2.2-1.8,4-4,4H9C6.8,42,5,40.2,5,38z"
          fill="#CFD8DC"
        />
        <path
          d="M43,10v6H5v-6c0-2.2,1.8-4,4-4h30C41.2,6,43,7.8,43,10z"
          fill="#F44336"
        />
        <g
          fill="#B71C1C"
        >
          <circle
            cx="33"
            cy="10"
            r="3"
          />
          <circle
            cx="15"
            cy="10"
            r="3"
          />
        </g>
        <g
          fill="#B0BEC5"
        >
          <path
            d="M33,3c-1.1,0-2,0.9-2,2v5c0,1.1,0.9,2,2,2s2-0.9,2-2V5C35,3.9,34.1,3,33,3z"
          />
          <path
            d="M15,3c-1.1,0-2,0.9-2,2v5c0,1.1,0.9,2,2,2s2-0.9,2-2V5C17,3.9,16.1,3,15,3z"
          />
        </g>
        <g
          fill="#90A4AE"
        >
          <rect
            height="4"
            width="4"
            x="13"
            y="20"
          />
          <rect
            height="4"
            width="4"
            x="19"
            y="20"
          />
          <rect
            height="4"
            width="4"
            x="25"
            y="20"
          />
          <rect
            height="4"
            width="4"
            x="31"
            y="20"
          />
          <rect
            height="4"
            width="4"
            x="13"
            y="26"
          />
          <rect
            height="4"
            width="4"
            x="19"
            y="26"
          />
          <rect
            height="4"
            width="4"
            x="25"
            y="26"
          />
          <rect
            height="4"
            width="4"
            x="31"
            y="26"
          />
          <rect
            height="4"
            width="4"
            x="13"
            y="32"
          />
          <rect
            height="4"
            width="4"
            x="19"
            y="32"
          />
          <rect
            height="4"
            width="4"
            x="25"
            y="32"
          />
          <rect
            height="4"
            width="4"
            x="31"
            y="32"
          />
        </g>
      </svg>
       
      <span>
        Book Now
      </span>
    </button>
  </span>
</section>
`;
